### PR TITLE
Code sonar cleanup

### PR DESCRIPTION
**Pull request by bandi13**

### CODE REVIEW

File: src/ssl_load.c [hunk 1]
```diff
@@ -5102,6 +5102,22 @@ int wolfSSL_CTX_use_RSAPrivateKey(WOLFSSL_CTX* ctx, WOLFSSL_RSA* rsa)
 
 #ifdef OPENSSL_EXTRA
 
+static char* wolfSSL_getenv_memcpy(const char* varName) {
```
Comment (bandi13) [inline src/ssl_load.c:5105]: Is there a better place for this function?

File: src/ssl_load.c [hunk 2]
```diff
@@ -5178,6 +5196,14 @@ int wolfSSL_CTX_set_default_verify_paths(WOLFSSL_CTX* ctx)
     #endif
     }
 
+#ifdef XGETENV
+    if (certFile != NULL) {
```
Comment (SparkiDev) [inline src/ssl_load.c:5200]: XFREE checks for NULL - no need to do it here.

File: src/ssl_load.c (hunk 1 above)
Comment (SparkiDev) [reply src/ssl_load.c:5106]: You could wrap the call to XGETENV with a duplicate string call.
That is XSTRDUP but using XMALLOC. Which I guess can go into wc_port.c???

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl_load.c (hunk 2 above)
Comment (bandi13) [reply src/ssl_load.c:5200]: I don't think that's true or else why would we do [this](https://github.com/wolfSSL/wolfssl/blob/master/linuxkm/lkcapi_glue.c#L2454) among other places in the code. [Here](https://github.com/wolfSSL/wolfssl/blob/master/wolfcrypt/src/memory.c#L414) is an example where we don't check to make sure it's not NULL. It is used when WOLFSSL_XFREE_NO_NULLNESS_CHECK is defined.

**Comment by bandi13** [comment]
Retest this please

File: src/ssl_load.c (hunk 2 above)
Comment (SparkiDev) [reply src/ssl_load.c]: *sigh* this is one that has been going on for a long time.
At one time we changed all the XFREE macros to check for NULL before calling free or whatever for the platform.
I'm not sure but I think WOLFSSL_XFREE_NO_NULLNESS_CHECK is used to remove the check as the free implementation does it.

File: wolfssl/wolfcrypt/types.h [hunk 3]
```diff
@@ -917,6 +917,7 @@ typedef struct w64wrapper {
         WOLFSSL_API int wc_strncasecmp(const char *s1, const char *s2, size_t n);
     #endif
 
+    WOLFSSL_API char* wc_strdup(const char *src, int memType);
```
Comment (dgarske) [inline wolfssl/wolfcrypt/types.h:920]: This should not be public... Please make WOLFSSL_LOCAL

File: wolfcrypt/src/wc_port.c [hunk 4]
```diff
@@ -1177,6 +1177,21 @@ int wc_strncasecmp(const char *s1, const char *s2, size_t n)
 }
 #endif /* USE_WOLF_STRNCASECMP */
 
+char* wc_strdup(const char *src, int memType) {
```
Comment (dgarske) [inline wolfcrypt/src/wc_port.c:1180]: This should not be always exposed... It needs to be conditional based on `#ifdef XGETENV` or possibly another macro like `USE_WOLF_STRDUP`. I'd like to see this strdup function perfectly match the stdlib strdup API. Also consider fixing `#define BUF_strdup strdup` to use it...

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by SparkiDev — APPROVED** [review]

File: wolfcrypt/src/wc_port.c (hunk 4 above)
Comment (bandi13) [reply wolfcrypt/src/wc_port.c]: Done

File: wolfssl/wolfcrypt/types.h (hunk 3 above)
Comment (bandi13) [reply wolfssl/wolfcrypt/types.h]: Done

**Review by dgarske — APPROVED** [review]

**Pull request merged**